qucs/qucs/components/componentdialog.cpp

```diff
@@ -602,7 +602,8 @@ void ComponentDialog::slotSelectProperty(QTableWidgetItem *item)
     ButtAdd->setEnabled(true);
     ButtRem->setEnabled(true);
 
-    if (Comp->Description == "equation") {
+    // enable Up/Down buttons only for the Equation component
+    if (Comp->Model == "Eqn") {
```

Comment:
sure. first of all, move the equation code out of componentdialog. it's clearly in the wrong place, regardless of what it does. a better way would be to have the equation code in the equation "component".

the `Description` and `Model` strings are a misconception, related to the cause of this bug. switching one for the other is not an improvement.

if fixing (including reviewing the fix) this bug is harder than cleaning up, then certainly a better way is to clean up.

just put `Comp->doSomethingWithDialog(this);`  instead. choose a proper name.